core/src/main/java/com/novoda/noplayer/internal/exoplayer/drm/ProvisioningModularDrmCallback.java

```diff
@@ -25,7 +28,13 @@
     }
 
     @Override
-    public byte[] executeKeyRequest(UUID uuid, ExoMediaDrm.KeyRequest request) throws Exception {
-        return streamingModularDrm.executeKeyRequest(new ModularDrmKeyRequest(request.getDefaultUrl(), request.getData()));
+    public byte[] executeKeyRequest(UUID uuid, ExoMediaDrm.KeyRequest request, @Nullable String mediaProvidedLicenseServerUrl) throws Exception {
+        String url = request.getDefaultUrl();
+
+        if (TextUtils.isEmpty(url)) {
+            url = mediaProvidedLicenseServerUrl;
```

Comment:
is this ok to be null?

Reply:
Rather interestingly I don't think this url is ever used by us. We actually expect a client to pass in the Url as part of the `DrmHandler` because it's a `no-player` class rather than an `ExoPlayer` class.